### PR TITLE
Fix: covert video and implement browser and os info for test session

### PR DESCRIPTION
**Pull request by emarf**
Wrote custom converter from webm to mp4
Changed logic of creation test sessions
Implemented browser and os settings with help of @ua-parser-js library
Screenshot 2022-01-12 at 17 09 19
Screenshot 2022-01-12 at 17 10 22

### CODE REVIEW

File: src/lib/zebReporter.ts [hunk 1]
```diff
@@ -284,53 +285,57 @@ class ZebRunnerReporter implements Reporter {
...
+        
+        return res;
       });
-    const result = results.flat();
 
-    return {results: result, errors};
+    return {results, errors};
   }
 
+  // ? mb remove this
```
Comment (ryanrosello-og) [inline src/lib/zebReporter.ts]: yup, I agree

File: src/lib/ResultsParser.ts [hunk 2]
```diff
@@ -123,56 +154,57 @@ export default class ResultsParser {
...
+        suite.tests,
+        launchInfo
       );
       this.updateResults({
         tests: testResults,
       });
       return;
     }
   }
+  // ? mb remove this
```
Comment (ryanrosello-og) [inline src/lib/ResultsParser.ts]: yup, I agree - lets remove

File: tests/pw_parser.spec.ts [hunk 3]
```diff
@@ -140,5 +171,5 @@ test('parses test details from the results @unit_test', async ({parsedResults})
   expect(parsedResults.tests[0].endedAt).toEqual('2021-12-18T09:43:23.630Z');
   expect(parsedResults.tests[0].name).toEqual('nested foo > foo - L2 > basic test @broke');
   expect(parsedResults.tests[0].suiteName).toEqual('nested foo > foo - L2');
-  expect(parsedResults.tests[0].browser).toBe(undefined);
+  expect(parsedResults.tests[0].browserCapabilities).toBe(undefined);
```
Comment (ryanrosello-og) [inline tests/pw_parser.spec.ts]: should this still be undefined? it'd be good to have at least one 1 test that covers the `browserCapabilities`

**Comment by ryanrosello-og** [comment]
looks good so far :)

File: tests/pw_parser.spec.ts (hunk 3 above)
Comment (emarf) [reply tests/pw_parser.spec.ts]: okay, i fix:)

**Comment by emarf** [comment]
Added browserCapabilities check in test
Remove useless code
Change session start and end date
Screenshot 2022-01-14 at 13 52 26

**Pull request merged**